README.md

```diff
@@ -203,6 +214,76 @@ you are recommended to take a look at documentation for the `holochain_wind_tunn
+
+Writing a Kitsune Wind Tunnel scenario is relatively straight forward. The Kitsune client defines three common functions for the developer. A chatter can be `create`d, it can `join_chatter_network` and it can `say` a list of messages. As long as a chatter has not joined the network, it won't receive messages from other peers and will also not send messages to them. Once joined, it starts receiving and sending messages it has said.
+
+For communication among peers to work, a bootstrap server must be running that enables peers to discover each other, and a signal server is required for establishing direct WebRTC connections. See [Kitsune Tests](#kitsune-tests).
+
+The only Wind Tunnel specific dependency you will need is `kitsune_wind_tunnel_runner`.
+
+```bash
+cargo add kitsune_wind_tunnel_runner
+```
+
+If this scenario is being written inside this repository then there are some extra setup steps. Please see the [project layout docs](#navigating-the-project).
+
+Add the following imports to the top of your `main.rs`:
+
+```rust
+use kitsune_wind_tunnel_runner::prelude::*;
+```
+
+Then replace your `main` function with the following:
+
+```rust
+fn main() -> WindTunnelResult<()> {
+    let builder =
+        KitsuneScenarioDefinitionBuilder::<KitsuneRunnerContext, KitsuneAgentContext>::new_with_init(
+            "scenario_name",
+        )?.into_std()
+        .use_agent_behaviour(agent_behavior);
+
+    run(builder)?;
+
+    Ok(())
+}
+```
+
+This is the basic structure of a Kitsune Wind Tunnel scenario. The `ScenarioDefinitionBuilder` is used to define the scenario. It includes
+a CLI which will allow you to override some of the defaults that are set in your code. Using the builder you can configure your hooks
+which are just Rust functions that take a context and return a `WindTunnelResult`. 
+
+The `run` function is then called with the builder. At that point the Wind Tunnel runner takes over and configures, then runs your scenario.
+
+Before you can run this, you'll need to provide the agent behaviour hook. Add the following to your `main.rs`:
+
+```rust
+fn agent_behaviour(
+    ctx: &mut AgentContext<KitsuneRunnerContext, KitsuneAgentContext>,
+) -> HookResult {
+    println!("Hello from chatter {}", ctx.get().get_chatter_id());
+    std::thread::sleep(std::time::Duration::from_secs(1));
+    Ok(())
+}
+```
+
+This is just an example hook and you will want to replace it once you have got your scenario running. Note the `KitsuneAgentContext` that is provided
+to the hook. This is created per-agent and gives you access to the agent's ID and the runner's context as well as the chatter ID which is specific to
+Kitsune. Both the agent and the runner context are used for sharing configuration between the runner and your hooks, and state between your hooks.
+
+Your scenario should now be runnable. Try running it with
```

Comment:
Will it run without a setup that joins the network?

Reply:
Ah no, it won't. I was really too lazy to actually try this out!

Reply:
Nice, yes that's good